apps/src/bin/quiche-server.rs

```diff
@@ -326,289 +341,383 @@ fn main() {
+                    // it was given to
+                    clients_ids.insert(preferred_address_scid, client_id);
+
+                    // Create the new connection!
+                    #[allow(unused_mut)]
+                    // Automatically provisions SCID for preferred address.
+                    let mut conn = quiche::accept(
+                        &scid,
+                        odcid.as_ref(),
+                        local_addr,
+                        from,
+                        &mut config,
+                    )
+                    .unwrap();
 
-                clients.insert(client_id, client);
-                clients_ids.insert(scid.clone(), client_id);
+                    if let Some(keylog) = &mut keylog {
+                        if let Ok(keylog) = keylog.try_clone() {
+                            conn.set_keylog(Box::new(keylog));
+                        }
+                    }
 
-                next_client_id += 1;
+                    // Only bother with qlog if the user specified it.
+                    #[cfg(feature = "qlog")]
+                    {
+                        if let Some(dir) = std::env::var_os("QLOGDIR") {
+                            let id = format!("{:?}", &scid);
+                            let writer = make_qlog_writer(&dir, "server", &id);
+
+                            conn.set_qlog(
+                                std::boxed::Box::new(writer),
+                                "quiche-server qlog".to_string(),
+                                format!("{} id={}", "quiche-server qlog", id),
+                            );
+                        }
+                    }
 
-                clients.get_mut(&client_id).unwrap()
-            } else {
-                let cid = match clients_ids.get(&hdr.dcid) {
-                    Some(v) => v,
+                    let client = Client {
+                        conn,
+                        http_conn: None,
+                        client_id,
+                        partial_requests: HashMap::new(),
+                        partial_responses: HashMap::new(),
+                        app_proto_selected: false,
+                        max_datagram_size,
+                        loss_rate: 0.0,
+                        max_send_burst: MAX_BUF_SIZE,
+                    };
 
-                    None => clients_ids.get(&conn_id).unwrap(),
-                };
+                    // Track the new client
+                    clients.insert(client_id, client);
+                    clients_ids.insert(scid.clone(), client_id);
 
-                clients.get_mut(cid).unwrap()
-            };
+                    next_client_id += 1;
 
-            let recv_info = quiche::RecvInfo {
-                to: local_addr,
-                from,
-            };
+                    clients.get_mut(&client_id).unwrap()
+                } else {
+                    trace!("We've seen this connection before!");
```

Comment:
seems quite spammy

Reply:
Updated as mentioned above. It is not too bad. Only shows up a couple of times.